### PR TITLE
[dua] add `IsDomainUnicast()` helper method

### PR DESCRIPTION
**Pull request by librasungirl**

### CODE REVIEW

File: src/core/backbone_router/leader.cpp [hunk 1]
```diff
@@ -288,6 +289,21 @@ void Leader::UpdateDomainPrefixConfig(void)
 #endif
 }
 
+bool Leader::IsOnDomain(const Ip6::Address &aAddress) const
+{
+    bool rval = false;
```
Comment (bukepo) [inline src/core/backbone_router/leader.cpp:294]: return HasDomainPrefix() && aAddress.PrefixMatch(mDomainPrefix.mPrefix) >= mDomainPrefix.mLength;

Comment (bukepo) [inline src/core/backbone_router/leader.cpp:294]: return HasDomainPrefix() && aAddress.PrefixMatch(mDomainPrefix.mPrefix) >= mDomainPrefix.mLength;

File: src/core/backbone_router/leader.hpp [hunk 2]
```diff
@@ -152,6 +152,16 @@ class Leader : public InstanceLocator
...
 
+    /**
+     * This method indicates whether or not the address is a Domain Unicast Address.
+     *
+     * @param[in]  aAddress A reference to the address.
+     *
+     * @retval true if @p aAddress is a domain unicast address, false otherwise.
+     *
+     */
+    bool IsOnDomain(const Ip6::Address &aAddress) const;
```
Comment (jwhui) [inline src/core/backbone_router/leader.hpp:163]: nit: how about naming something like `IsDomainUnicast`?

File: src/core/backbone_router/leader.hpp [hunk 3]
```diff
@@ -152,6 +152,16 @@ class Leader : public InstanceLocator
      */
     bool HasDomainPrefix(void) const { return (mDomainPrefix.mLength > 0); }
 
+    /**
+     * This method indicates whether or not the address is a Domain Unicast Address.
+     *
+     * @param[in]  aAddress A reference to the address.
+     *
+     * @retval true if @p aAddress is a domain unicast address, false otherwise.
```
Comment (jwhui) [inline src/core/backbone_router/leader.hpp:160]:
```suggestion
     * @retval true  @p aAddress is a domain unicast address.
     * @retval false @p aAddress is not a domain unicast address.
```

File: src/core/backbone_router/leader.cpp [hunk 4]
```diff
@@ -35,6 +35,7 @@
 
 #if (OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2)
 
+#include "common/code_utils.hpp"
```
Comment (bukepo) [inline src/core/backbone_router/leader.cpp:38]:
```suggestion
```

**Pull request merged**